pkg/kubelet/kubelet_volumes.go

```diff
@@ -63,6 +64,10 @@ func (kl *Kubelet) podVolumesExist(podUID types.UID) bool {
 	volumePaths, err := kl.getMountedVolumePathListFromDisk(podUID)
 	if err != nil {
 		klog.Errorf("pod %q found, but error %v occurred during checking mounted volumes from disk", podUID, err)
+		if mount.IsCorruptedMnt(err) {
+			klog.Errorf("pod %q: found a corrupted mount point with error(%v), mark this volume as not exist", podUID, err)
+			return false
+		}
```

Comment:
I am not sure if this issue should be fixed here. We made a similar fix for flexvolume's here - https://github.com/kubernetes/kubernetes/pull/67097 . If mount point is known to be corrupt and since flexvolume assumes corrupt mounts to be "unmounted" but still calls `unmount` command, can the plugin not do cleanup and allow kubelet to remove corrupted mount point?

Reply:
hi @gnufied current issue is if that mount dir is corrupted, it won't go to `Unmount` process, it failes ealier at `syncPod --> canBeDeleted--> PodResourcesAreReclaimed --> podVolumesExist --> getMountedVolumePathListFromDisk --> IsLikelyNotMountPoint`, and the pod with corrupt mount can never be deleted any more.

Reply:
@andyzhangx the Unmount operation does not go through that path, it goes through volumeManager.  `syncPod` and volumeManager are two different threads and `syncPod` waits for volumeManager to finish unmounting (via the podVolumesExist check).  So if you're seeing that the volume directory still exists after some time, then that means something in the volume plugin `TearDown` or volumeManager is not correct.

Reply:
@msau42 as I could confirm, when smb path is corrupted, it won't go to `TearDown` func in https://github.com/kubernetes/kubernetes/blob/6ec5a7d3371729a6d922f66ad8703de5e3d6ac95/pkg/volume/flexvolume/unmounter.go#L44
while if smb path is correct, it will go to `TearDown` func.

Reply:
and I only found these two error logs:
```
Mar 11 15:03:01 k8s-agentpool-34398540-1 kubelet[27330]: E0311 15:03:01.118791   27330 kubelet_volumes.go:66] pod "2c4797d3-440e-11e9-9696-000d3a004ffb" found, but error stat /var/lib/kubelet/pods/2c4797d3-440e-11e9-9696-000d3a004ffb/volumes/microsoft.com~smb/test: permission denied occurred during checking mounted volumes from disk
Mar 11 15:03:01 k8s-agentpool-34398540-1 kubelet[27330]: W0311 15:03:01.118807   27330 mount_helper.go:125] IsCorruptedMnt failed with error: stat /var/lib/kubelet/pods/2c4797d3-440e-11e9-9696-000d3a004ffb/volumes/microsoft.com~smb/test: permission denied
```

So when I worked out this PR, it works, not sure which part is wrong in the root cause.

Reply:
It may be worth chasing down why `Teardown` is not getting called because volume mount being corrupted has nothing to do with `Teardown` not being called (and if it does it is a bug we should fix). 

This PR kinda fixes the problem in a roundabout way by leaving the mount point intact and just returning false from here. IMO - it isn't a great solution and we should be careful about making changes that affects all volume types.

Reply:
@andyzhangx you mentioned, when smb is corrupted, it won't go to TearDown func. Then could you see where it goes from the log? Any other volume operations being processed?

Reply:
thanks all for the suggestion, I have found the root cause, there is one error message not caught in `IsCorruptedMnt`!